### PR TITLE
Turn kinds in python API into a proper Enum

### PR DESCRIPTION
**Pull request by nafur**
This PR does multiple things:
- the kinds are changed from custom objects to a proper `enum.Enum` class (including according changes to the cython code and the kind generation scripts)
- all examples and tests are modified to account for the change how to use kinds (`Kind` instead of `kinds`)
- add docstrings to the kind enum values
- add a custom documenter that properly renders enums via autodoc
- extend doxygen setup so that we can write comments as `rst` (allowing us to copy the documentation for kinds from the cpp api to the other apis)

### CODE REVIEW

File: docs/ext/autoenum.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        # overriding this flag prints __doc__ just as we want to.
+        self.doc_as_attr = False
+        super().add_content(more_content, no_docstring)
+        self.doc_as_attr = True
+
+
+def setup(app: Sphinx) -> None:
+    app.setup_extension('sphinx.ext.autodoc')
+    app.add_autodocumenter(EnumDocumenter)
```
Comment (yoni206) [inline docs/ext/autoenum.py:41]:
```suggestion
    app.add_autodocumenter(EnumDocumenter)
    
```

File: examples/api/python/id.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 ##
 
 import pycvc5
-from pycvc5 import kinds
+from pycvc5 import Kind
```
Comment (yoni206) [inline examples/api/python/id.py:19]: Maybe this can even be deleted?

**Review by yoni206 — APPROVED** [review]
LGTM, modulo minor comments inline.

**Comment by yoni206** [comment]
added do-not-merge to allow a review from @aniemetz before merging.

File: docs/api/python/regular/python.rst [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Python API Documentation
+regular Python API
```
Comment (aniemetz) [inline docs/api/python/regular/python.rst:1]:
```suggestion
Regular Python API Documentation
```

File: docs/api/python/python.rst [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Python API Documentation
+Python API
```
Comment (aniemetz) [inline docs/api/python/python.rst:1]: Should the visibility of the individual api toctrees really be hidden?
Also: `compatibility API` -> `Compatibility Python API` and `regular Python API` -> `Regular Python API`
Can you make the former change also for the header of the compat api page?

**Review by aniemetz — CHANGES_REQUESTED** [review]

**Review by aniemetz — APPROVED** [review]

**Pull request merged**